# 3-say_my_name.txt

"""
::
>>> say_my_name = __import__('3-say_my_name').say_my_name

::
Case of Both Names are supplied
>>> say_my_name("Adeyemi", "Samuel")
My name is Adeyemi Samuel

::
Case of Only one name is supplied
>>> say_my_name("Adeyemi")
My name is Adeyemi

::
Case of no name is supplied
>>> say_my_name()
Traceback (most recent call last):
	...
TypeError: say_my_name() missing 1 required positional argument: 'first_name'

::
Case of only one str and another data type is supplied
>>> say_my_name("Adeyemi", 5)
Traceback (most recent call last):
	...
TypeError: last_name must be a string

::
Case of only one str and another data type is supplied
>>> say_my_name(5, "Adeyemi")
Traceback (most recent call last):
	...
TypeError: first_name must be a string

::
Case of only one str and another data type is supplied
>>> say_my_name(7.5, "Samuel")
Traceback (most recent call last):
	...
TypeError: first_name must be a string

::
Case of only one str and another data type is supplied
>>> say_my_name("Samuel", 7.5)
Traceback (most recent call last):
	...
TypeError: last_name must be a string

::
Case of two str supplied
>>> say_my_name("Samuel", "7.5")
My name is Samuel 7.5

::
Case of two str supplied
>>> say_my_name("Samuel", "75")
My name is Samuel 75

::
Case of only a number is supplied
>>> say_my_name(5)
Traceback (most recent call last):
	...
TypeError: first_name must be a string

::
Case of a float is supplied
>>> say_my_name(5.5)
Traceback (most recent call last):
	...
TypeError: first_name must be a string

::
Case of two floats are supplied
>>> say_my_name(1.5, 2.5)
Traceback (most recent call last):
	...
TypeError: first_name must be a string

::
Case of two integers are supplied
>>> say_my_name(1.5, 2.5)
Traceback (most recent call last):
	...
TypeError: first_name must be a string

::
Case of a float and an integer
>>> say_my_name(1.5, 25)
Traceback (most recent call last):
	...
TypeError: first_name must be a string

::
Case of an integer and a float
>>> say_my_name(12, 1.2)
Traceback (most recent call last):
	...
TypeError: first_name must be a string

::
Case of too many arguments
>>> say_my_name("Adeyemi", "Samuel", "Adewale")
Traceback (most recent call last):
	...
TypeError: say_my_name() takes from 1 to 2 positional arguments but 3 were given

"""
